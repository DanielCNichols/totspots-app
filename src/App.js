import './app.css';
import React, { Component } from 'react';
import { Route, Switch } from 'react-router-dom';
import VenuesContext from './VenuesContext';
import Landingpage from './LandingPage/Landingpage';
import ErrorBoundary from './ErrorBoundary/ErrorBoundary';
import Sidebar from './Sidebar/Sidebar';
import Login from './Login/Login';
import Registration from './Registration/Registration';
import RefineSearch from './RefineSearch/RefineSearch';
import ResultsPage from './ResultsPage/ResultsPage';
import Nav from './Nav/Nav';
import Main from './Main/Main';
import ProfileView from './Profileview/Profileview';
import AddVenue from './AddVenue/AddVenue';
import ReviewForm from './ReviewForm/ReviewForm';
import VenuesPage from './VenuesPage/VenuesPage';
import FavoritesList from './FavoritesList/Favoriteslist';
import editReview from './EditReview/EditReview';
import PublicRoute from './utils/PublicRoute';
import PrivateRoute from './utils/PrivateRoute';
import Footer from './Footer/Footer';
import ReviewSort from './ReviewSort/ReviewSort';

export default class App extends Component {
  static ContextType = VenuesContext;

  componentDidMount() {
    document.title = 'Totspots';
  }

  render() {
    return (
      <div className="app">
        <ErrorBoundary>
          <Route path="/" component={Nav} />
          <Sidebar>
            <Switch>
              <Route
                exact
                path="/reviews/:city/:queryState/:type"
                component={RefineSearch}
              />
              <Route exact path="/venue/:venue_id" component={ReviewSort} />
            </Switch>
          </Sidebar>
          <Main className="app_main">
            <Switch>
              <Route exact path="/" component={Landingpage} />
              <PublicRoute exact path="/login" component={Login} />
              <PublicRoute exact path="/register" component={Registration} />
              <Route
                exact
                path="/reviews/:city/:queryState/:type"
                component={ResultsPage}
              />
              <PrivateRoute exact path="/account" component={ProfileView} />
              <PrivateRoute exact path="/account" component={FavoritesList} />
              <PrivateRoute exact path="/addVenue" component={AddVenue} />
              <Route exact path="/venue/:venue_id" component={VenuesPage} />
              <PrivateRoute
                exact
                path="/addReview/:venue_id"
                component={ReviewForm}
              />
              <PrivateRoute
                exact
                path="/reviews/:review_id"
                component={editReview}
              />
            </Switch>
          </Main>
          <footer className="app_footer">
            <Switch>
              <Route path="/" component={Footer} />
            </Switch>
          </footer>
        </ErrorBoundary>
      </div>
    );
  }
}
