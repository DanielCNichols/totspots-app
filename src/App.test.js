
import React from 'react';
import ReactDOM from 'react-dom'
import App from './App.js'
import {MemoryRouter} from 'react-router-dom'
import renderer from 'react-test-renderer'



it('renders App without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(
  <MemoryRouter>
    <App />
  </MemoryRouter>, div);
  ReactDOM.unmountComponentAtNode(div)
})

it('renders App as expected', () => {
  const tree = renderer
  .create(
  <MemoryRouter>
    <App name="App"/>
  </MemoryRouter>
  )
  .toJSON();
  expect(tree).toMatchSnapshot();
})
