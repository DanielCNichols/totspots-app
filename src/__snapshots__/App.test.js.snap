// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders App as expected 1`] = `
<div
  className="app"
>
  <nav>
    <ul>
      <li>
        <a
          href="/"
          onClick={[Function]}
        >
          Home
        </a>
      </li>
      <li>
        <a
          href="/addVenue"
          onClick={[Function]}
        >
          Add Venue
        </a>
      </li>
      <li
        className="Nav_logout"
      >
        <a
          href="/login"
          onClick={[Function]}
        >
          Log in
        </a>
      </li>
    </ul>
  </nav>
  <div
    className="sidebar"
  >
    <div>
      Hello World
    </div>
  </div>
  <div
    className="main"
  >
    <section
      className="landingpage"
    >
      <header
        className="landingpage_hero"
      >
        <h1>
          Tot Spots
        </h1>
        <p>
          For parents, by parents.
        </p>
      </header>
      <div
        className="search"
      >
        <p>
          I want to take my family to a...
        </p>
        <form
          className="search_form"
          onSubmit={[Function]}
        >
          <label
            className="search_form_label"
            htmlFor="type"
            required={true}
          >
            Venue or event...
            <select
              className="search_form_select"
              id="type"
              name="type"
            >
              <option
                value=""
              >
                Select a type of venue/event
              </option>
              <option
                value="Restaurant"
              >
                Restaurants
              </option>
              <option
                value="Bar"
              >
                Brewery/Bar
              </option>
              <option
                value="Educational"
              >
                Educational
              </option>
              <option
                value="music"
              >
                Music
              </option>
              <option
                value="outdoor"
              >
                Outdoor
              </option>
            </select>
          </label>
          <p>
            in...
          </p>
          <label
            className="search_form_label"
            htmlFor="city"
          >
             City
            <input
              className="search_form_text"
              id="city"
              name="city"
              placeholder="Durham"
              required={true}
              type="text"
            />
          </label>
          <label
            className="search_form_label"
            htmlFor="state"
          >
             State
            <input
              className="search_form_text"
              max-length="2"
              name="state"
              placeholder="NC"
              required={true}
              type="text"
            />
          </label>
          <div
            className="search_form_control"
          >
            <button
              className="search_form_button"
              type="submit"
            >
              Let's go!
            </button>
          </div>
        </form>
      </div>
      <div
        role="alert"
      />
      <p>
        Have an account? 
        <a
          href="/login"
          onClick={[Function]}
        >
          Log in
        </a>
         or
        <a
          href="/register"
          onClick={[Function]}
        >
           sign up
        </a>
      </p>
    </section>
  </div>
  <footer
    className="app_footer"
  >
    <div
      className="footer"
    >
      <p>
        Daniel Nichols | 2019
      </p>
      <a
        href="https://github.com/DanielCNichols"
      >
        <p>
          Want to see more? Find it on GitHub
        </p>
      </a>
    </div>
  </footer>
</div>
`;
